Cabal-syntax/Cabal-syntax.cabal

```diff
@@ -0,0 +1,243 @@
+cabal-version: >=1.10
+name:          Cabal-syntax
+version:       3.7.0.0
+copyright:     2003-2021, Cabal Development Team (see AUTHORS file)
+license:       BSD3
+license-file:  LICENSE
+author:        Cabal Development Team <cabal-devel@haskell.org>
+maintainer:    cabal-devel@haskell.org
+homepage:      http://www.haskell.org/cabal/
+bug-reports:   https://github.com/haskell/cabal/issues
+synopsis:      A library for working with .cabal files
+description:
+    This library provides tools for reading and working with the .cabal file
+    format.
+category:       Distribution
+build-type:     Simple
+
+extra-source-files:
+  README.md ChangeLog.md
+
+source-repository head
+  type:     git
+  location: https://github.com/haskell/cabal/
+  subdir:   Cabal-syntax
+
+flag bundled-binary-generic
+  default: False
+
+library
+  default-language: Haskell2010
+  hs-source-dirs: src
+
+  build-depends:
+    array      >= 0.4.0.1  && < 0.6,
+    base       >= 4.6      && < 5,
+    bytestring >= 0.10.0.0 && < 0.12,
+    containers >= 0.5.0.0  && < 0.7,
+    deepseq    >= 1.3.0.1  && < 1.5,
+    directory  >= 1.2      && < 1.4,
+    filepath   >= 1.3.0.1  && < 1.5,
+    pretty     >= 1.1.1    && < 1.2,
+    time       >= 1.4.0.1  && < 1.12
+
+  if flag(bundled-binary-generic)
+    build-depends: binary >= 0.5.1.1 && < 0.7
+  else
+    build-depends: binary >= 0.7 && < 0.9
+
+  if os(windows)
+    build-depends: Win32 >= 2.3.0.0 && < 2.13
+  else
+    build-depends: unix  >= 2.6.0.0 && < 2.8
+
+  ghc-options: -Wall -fno-ignore-asserts -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates
+  if impl(ghc >= 8.0)
+    ghc-options: -Wcompat -Wnoncanonical-monad-instances
+
+    if impl(ghc <8.8)
+      ghc-options: -Wnoncanonical-monadfail-instances
+
+  if !impl(ghc >= 8.0)
+    -- at least one of lib:Cabal's dependency (i.e. `parsec`)
+    -- already depends on `fail` and `semigroups` transitively
+    build-depends: fail == 4.9.*, semigroups >= 0.18.3 && < 0.20
+
+  if !impl(ghc >= 7.10)
+    build-depends: void >= 0.7.3 && < 0.8
+
+  if !impl(ghc >= 7.8)
+    -- semigroups depends on tagged.
+    build-depends:  tagged >=0.8.6 && <0.9, bytestring-builder >= 0.10.8 && <0.11
+
+  exposed-modules:
+    Distribution.CabalSpecVersion
+    Distribution.Compat.Binary
+    Distribution.Compat.DList
+    Distribution.Compat.Exception
+    Distribution.Compat.Lens
+    Distribution.Compat.Newtype
+    Distribution.Compat.NonEmptySet
+    Distribution.Compat.Prelude
+    Distribution.Compat.Semigroup
```

Comment:
We may want to split the compat stuff between Cabal and Cabal-syntax, like it's done now with cabal-install and Cabal, but I'm not sure it's worth doing in this case.

Reply:
I'd leave them for now, and just drop pre-GHC-8 support (#7531) before next major. That would cleanup plenty of stuff, leaving mostly "We rather not depend on `dlist` package and make our own copy".